rust/arrow/src/error.rs

```diff
@@ -24,6 +24,9 @@ use std::error::Error;
 /// Many different operations in the `arrow` crate return this error type.
 #[derive(Debug)]
 pub enum ArrowError {
+    /// "Not yet implemented" Arrow error.
+    /// Returned when functionality is not yet available.
+    NYI(String),
```

Comment:
I think that the error name should be explicit, like all others. I can't understand what `NYI` means without having to go to the docs, and in an error message, the person often has no access to the docs (at least not in a 1 click thing). 

`NotYetImplemented`?

Reply:
I agree that `NotYetImplemented` is a better name